### ISSUE TITLE
Build issues with avx

### ISSUE THREAD

**Issue by smuzaffar** [cms-sw/cmssw#31851]
Building CMSSW with `-mavx` or `-mavx2 -mfma` or `-march=sandybridge` generates build error [a]. I think the issue is that `__SSE__` is always set https://github.com/cms-sw/cmssw/blob/c345719fd55032e6894f67a51d8b91374555ad86/DataFormats/Math/interface/SIMDVec.h#L6

@VinInn , can you please propose a check to avoid it? I tried [b] which allowed me to build for both sse3, avx2 and also for `non-x86_64` archs (aarch64, ppc64le) but I am not sure if this is the right fix

[a]
```
In file included from gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/immintrin.h:43,
                 from gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/x86intrin.h:48,
                 from DataFormats/Math/interface/SSEVec.h:7,
                 from DataFormats/Math/interface/SIMDVec.h:57,
                 from DataFormats/GeometryVector/interface/Basic3DVector.h:4,
                 from DataFormats/GeometryVector/interface/PV3DBase.h:4,
                 from DataFormats/GeometryVector/interface/Point3DBase.h:5,
                 from DataFormats/GeometryVector/interface/GlobalPoint.h:5,
                 from TrackingTools/AnalyticalJacobians/interface/AnalyticalCurvilinearJacobian.h:5,
                 from TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobian.cc:1:
gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/avx2intrin.h: In function '__m256d mathSSE::_mm256_cross_pd(__m256d, __m256d)':
gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/avx2intrin.h:913:1: error: inlining failed in call to always_inline '__m256i _mm256_xor_si256(__m256i, __m256i)': target specific option mismatch
 _mm256_xor_si256 (__m256i __A, __m256i __B)
 ^~~~~~~~~~~~~~~~
In file included from DataFormats/Math/interface/SIMDVec.h:57,
                 from DataFormats/GeometryVector/interface/Basic3DVector.h:4,
                 from DataFormats/GeometryVector/interface/PV3DBase.h:4,
                 from DataFormats/GeometryVector/interface/Point3DBase.h:5,
                 from DataFormats/GeometryVector/interface/GlobalPoint.h:5,
                 from TrackingTools/AnalyticalJacobians/interface/AnalyticalCurvilinearJacobian.h:5,
                 from TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobian.cc:1:
DataFormats/Math/interface/SSEVec.h:92:36: note: called from here
     return __m256d(_mm256_xor_si256(__m256i(ret), neg));
                    ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
In file included from gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/immintrin.h:43,
                 from gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/x86intrin.h:48,
                 from DataFormats/Math/interface/SSEVec.h:7,
                 from DataFormats/Math/interface/SIMDVec.h:57,
     In file included from gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/immintrin.h:43,
                 from gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/x86intrin.h:48,
                 from DataFormats/Math/interface/SSEVec.h:7,
                 from DataFormats/Math/interface/SIMDVec.h:57,
                 from DataFormats/GeometryVector/interface/Basic3DVector.h:4,
                 from DataFormats/GeometryVector/interface/PV3DBase.h:4,
                 from DataFormats/GeometryVector/interface/Point3DBase.h:5,
                 from DataFormats/GeometryVector/interface/GlobalPoint.h:5,
                 from TrackingTools/AnalyticalJacobians/interface/AnalyticalCurvilinearJacobian.h:5,
                 from TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobian.cc:1:
gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/avx2intrin.h: In function '__m256d mathSSE::_mm256_cross_pd(__m256d, __m256d)':
gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/avx2intrin.h:913:1: error: inlining failed in call to always_inline '__m256i _mm256_xor_si256(__m256i, __m256i)': target specific option mismatch
 _mm256_xor_si256 (__m256i __A, __m256i __B)
 ^~~~~~~~~~~~~~~~
In file included from DataFormats/Math/interface/SIMDVec.h:57,
                 from DataFormats/GeometryVector/interface/Basic3DVector.h:4,
                 from DataFormats/GeometryVector/interface/PV3DBase.h:4,
                 from DataFormats/GeometryVector/interface/Point3DBase.h:5,
                 from DataFormats/GeometryVector/interface/GlobalPoint.h:5,
                 from TrackingTools/AnalyticalJacobians/interface/AnalyticalCurvilinearJacobian.h:5,
                 from TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobian.cc:1:
DataFormats/Math/interface/SSEVec.h:92:36: note: called from here
     return __m256d(_mm256_xor_si256(__m256i(ret), neg));
                    ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
In file included from gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/immintrin.h:43,
                 from gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/x86intrin.h:48,
                 from DataFormats/Math/interface/SSEVec.h:7,
                 from DataFormats/Math/interface/SIMDVec.h:57,
                 from DataFormats/GeometryVector/interface/Basic3DVector.h:4,
                 from DataFormats/GeometryVector/interface/PV3DBase.h:4,
                 from DataFormats/GeometryVector/interface/Point3DBase.h:5,
                 from DataFormats/GeometryVector/interface/GlobalPoint.h:5,
                 from TrackingTools/AnalyticalJacobians/interface/AnalyticalCurvilinearJacobian.h:5,
                 from TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobian.cc:1:
gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/avx2intrin.h:913:1: error: inlining failed in call to always_inline '__m256i _mm256_xor_si256(__m256i, __m256i)': target specific option mismatch
 _mm256_xor_si256 (__m256i __A, __m256i __B)
 ^~~~~~~~~~~~~~~~
In file included from DataFormats/Math/interface/SIMDVec.h:57,
                 from DataFormats/GeometryVector/interface/Basic3DVector.h:4,
                 from DataFormats/GeometryVector/interface/PV3DBase.h:4,
                 from DataFormats/GeometryVector/interface/Point3DBase.h:5,
                 from DataFormats/GeometryVector/interface/GlobalPoint.h:5,
                 from TrackingTools/AnalyticalJacobians/interface/AnalyticalCurvilinearJacobian.h:5,
                 from TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobian.cc:1:
DataFormats/Math/interface/SSEVec.h:92:36: note: called from here
     return __m256d(_mm256_xor_si256(__m256i(ret), neg));
                    ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
gmake: *** [config/SCRAM/GMake/Makefile.rules:1687: tmp/slc7_amd64_gcc820/src/TrackingTools/AnalyticalJacobians/src/TrackingToolsAnalyticalJacobians/sandybridge/AnalyticalCurvilinearJacobian.cc.o] Error 1
            from DataFormats/GeometryVector/interface/Basic3DVector.h:4,
                 from DataFormats/GeometryVector/interface/PV3DBase.h:4,
                 from DataFormats/GeometryVector/interface/Point3DBase.h:5,
                 from DataFormats/GeometryVector/interface/GlobalPoint.h:5,
                 from TrackingTools/AnalyticalJacobians/interface/AnalyticalCurvilinearJacobian.h:5,
                 from TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobian.cc:1:
gcc/8.2.0-bcolbf/lib/gcc/x86_64-unknown-linux-gnu/8.4.0/include/avx2intrin.h:913:1: error: inlining failed in call to always_inline '__m256i _mm256_xor_si256(__m256i, __m256i)': target specific option mismatch
 _mm256_xor_si256 (__m256i __A, __m256i __B)
 ^~~~~~~~~~~~~~~~
In file included from DataFormats/Math/interface/SIMDVec.h:57,
                 from DataFormats/GeometryVector/interface/Basic3DVector.h:4,
                 from DataFormats/GeometryVector/interface/PV3DBase.h:4,
                 from DataFormats/GeometryVector/interface/Point3DBase.h:5,
                 from DataFormats/GeometryVector/interface/GlobalPoint.h:5,
                 from TrackingTools/AnalyticalJacobians/interface/AnalyticalCurvilinearJacobian.h:5,
                 from TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobian.cc:1:
DataFormats/Math/interface/SSEVec.h:92:36: note: called from here
     return __m256d(_mm256_xor_si256(__m256i(ret), neg));
                    ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
gmake: *** [config/SCRAM/GMake/Makefile.rules:1687: tmp/slc7_amd64_gcc820/src/TrackingTools/AnalyticalJacobians/src/TrackingToolsAnalyticalJacobians/sandybridge/AnalyticalCurvilinearJacobian.cc.o] Error 1
```

[b]
```
--- a/DataFormats/Math/interface/SIMDVec.h
+++ b/DataFormats/Math/interface/SIMDVec.h
@@ -3,8 +3,12 @@
 
 #if (defined(__CLING__) || defined(__MIC__) || defined(__NVCC__)) || (__BIGGEST_ALIGNMENT__ < 16)
 #elif defined(__GNUC__) || defined(__clang__) || defined(__INTEL_COMPILER)
-#if defined(__x86_64__) && defined(__SSE__)
+#if defined(__x86_64__)
+#if defined(__AVX__)
+#define USE_EXTVECT
+#else
 #define USE_SSEVECT
+#endif
 #else
 #define USE_EXTVECT
 #endif
```

**Comment by makortel** [comment]
assign core, reconstruction

**Comment by VinInn** [comment]
I lost track of all those options and why one cannot just "USE_EXTVECT".
The best is most probably to remove the AVX part from AnalyticalCurvilinearJacobian.cc

**Comment by smuzaffar** [comment]
+core
fixed by #33268
